examples/receipts.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright 2024 Fuel Labs <contact@fuel.sh>
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::Result;
+use fuel_streams::{core::prelude::SubjectBuildable, prelude::*, receipts::*};
+use futures::StreamExt;
+
+/// The URL of the Fuel streaming service.
+const FUEL_STREAMING_SERVICE_URL: &str = "nats://fuel-streaming.testnet:4222";
+
+/// The contract ID to stream the receipts for.
+const CONTRACT_ID: &str =
+    "0x0000000000000000000000000000000000000000000000000000000000000000";
+
+/// Subscribes to receipts related to a specific contract, effectively listening to contract events.
+///
+/// This function creates a stream that subscribes to various types of receipts (except `ScriptResult`
+/// and `MessageOut`) that are associated with the specified contract ID. It's a way to monitor
+/// contract-related events such as calls, returns, logs, transfers, mints, and burns.
+///
+/// The function filters the receipts to ensure they match the given contract ID before processing them.
+/// This approach allows for efficient monitoring of contract activities without the need to process
+/// irrelevant receipts.
+///
+/// # Arguments
+///
+/// * `client` - A reference to the NATS client used for streaming.
+/// * `contract_id` - The ID of the contract to monitor.
+///
+/// # Returns
+///
+/// Returns `Ok(())` if the streaming completes successfully, or an error if there are any issues.
+async fn stream_contract_events(
```

Comment:
please, add everything in a `main` function instead of call this other function, as much simple as possible